features/323-login-id/stage-3.md

```diff
@@ -0,0 +1,50 @@
+# User settings
+
+## Database Schema
+
+- Rename `_auth_user_profile` table to `_auth_user`
+    - Rename `data` column to `metadata`
+
+- Add `_auth_user.primary_principal_id` column (`TEXT`, foreign key to
+  `_auth_principal.id` column)
+
+
+## Configuration
+- Let `WelcomeEmailDestination` be an enum:
+    - `primary`: Primary login ID of user
+    - `all`: All supported login IDs of user
+- Update field `welcomeEmail`:
+    - Add field `destination` (enum `WelcomeEmailDestination`,
+      optional, default to `primary`)
+
+
+## HTTP API
+
+- `POST /signup`, `POST /login`, etc.
+    - Return the primary login ID (string or null). If primary principal is not
+      password authenticated, primary login ID is null.
+    - Example:
+      ```json
+      {
+          "user_id": "...",
+          "access_token": "...",
+          "primary_login_id": "kiootic@oursky.com"
+      }
+      ```
+
+
+## Auth logic
+
+- Signup:
+    - Set user's primary login ID to first login ID with key configured as
+      supported standard keys (`email`, `phone`) in `login_ids` list.
+    - If no supported login ID is found, set primary login ID to
+      first login ID in list.
+
+- Welcome Email:
+    - Send email to destinations specified in config:
```

Comment:
Maybe it can be specified by config, and if not specified, could be specified in functions (I didn't give much thought on it on use cases, so just say "maybe".... not sure)

Reply:
This would be out of scope for this document: we will need to revisit the welcome message feature in future anyways.

Reply:
ok. Please remove things that's out of scope. (Or make it clear it is just some "reference for future design"

Reply:
The issue want to address where to send the welcome email if user has >=2 emails, so all/primary is within scope.
However, specifying the welcome email destination as cloud function would be out of scope, since it likely involve full customization of welcome message sending logic.

Reply:
ok. We can keep it simple first.